diff --git a/3rd_party/aids-patched.hpp b/3rd_party/aids-patched.hpp
index 96500bc..8358093 100644
--- a/3rd_party/aids-patched.hpp
+++ b/3rd_party/aids-patched.hpp
@@ -124,8 +124,14 @@
 #include <cstdlib>
 #include <cstring>
 
+
+
+
+
+struct PersonID;
 namespace aids
 {
+    unsigned long hash(PersonID);
 ////////////////////////////////////////////////////////////
 // ALLOCATORS
 ////////////////////////////////////////////////////////////
