diff --git a/3rd_party/aids-patched.hpp b/3rd_party/aids-patched.hpp
index 96500bc..5ae5834 100644
--- a/3rd_party/aids-patched.hpp
+++ b/3rd_party/aids-patched.hpp
@@ -1128,6 +1128,20 @@ namespace aids
                 return {};
             }
         }
+
+        Value *operator[](Key key) {
+            {
+                Maybe<Value*> maybe_value = get(key);
+                if (not maybe_value.has_value) {
+                    insert(key, {});
+                } else {
+                    return maybe_value.unwrap;
+                }
+            }
+            Maybe<Value*> maybe_value = get(key);
+            assert(maybe_value.has_value);
+            return maybe_value.unwrap;
+        }
     };
 
     template <typename Key, typename Value>
